src/oas3/__tests__/operation.test.ts

```diff
@@ -570,4 +570,99 @@ describe('transformOas3Operation', () => {
       tags: [],
     });
   });
+
+  test('should handle server variables', () => {
+    const document: Partial<OpenAPIObject> = {
+      id: '?http-service-id?',
+      paths: {
+        '/pets': {
+          get: {
+            responses: {
+              '200': {
+                description: 'OK',
+              },
+            },
+            summary: 'List pets',
+          },
+        },
+      },
+      servers: [
+        {
+          url: 'https://petstore.swagger.io/v2',
+          description: 'Sample Petstore Server Https',
+          variables: {
+            username: {
+              default: 'demo',
+              description: 'value is assigned by the service provider',
+            },
+            port: {
+              enum: [8443, 443],
+              default: 8443,
+            },
+            basePath: {
+              default: 'v2',
+            },
+          },
+        },
+        {
+          url: 'http://petstore.swagger.io/v2',
+          description: 'Sample Petstore Server Http',
+        },
+      ],
+    };
+
+    expect(transformOas3Operation({ document, path: '/pets', method: 'get' })).toStrictEqual({
```

Comment:
Also, instead of testing the whole `transformOas3Operation`, we could just test the transform security function to make the test leaner. I'm ok with keeping this test though, but let's assert on the security object only

Reply:
Strict equality is something really difficult in this repo/situation.

```suggestion
    expect(transformOas3Operation({ document, path: '/pets', method: 'get' })).toEqual({
```